misc/ls-extensions/modules/partial-parser/src/main/java/io.ballerina.parsers/PartialParserService.java

```diff
@@ -110,4 +121,40 @@ private String getFormattedSource(String statement) {
 
         return formattedSourceCode;
     }
+
+    private String getFunctionBodiedFormattedSource(String statement) {
+
+        SyntaxTree syntaxTree = getSTForSourceWithinMain(statement);
+        String formattedSourceCode = statement;
+
+        try {
+            SyntaxTree formattedTree = Formatter.format(syntaxTree);
+            ModulePartNode modulePartNode = formattedTree.rootNode();
+            FunctionDefinitionNode functionDefinitionNode = (FunctionDefinitionNode) modulePartNode.members().get(0);
+            FunctionBodyBlockNode functionBodyBlockNode = (FunctionBodyBlockNode) functionDefinitionNode.functionBody();
+            formattedSourceCode = functionBodyBlockNode.statements().get(0).toSourceCode();
+        } catch (FormatterException | NullPointerException e) {
+            // TODO: Print a warn log in the language client.
```

Comment:
You can override the `init(LanguageServer langServer, WorkspaceManager workspaceManager, LanguageServerContext serverContext)` method and get the `LanguageClient` from the `LanguageServerContext`.

Reply:
Fixed with b72aafb4f96c38abee11e6f543123806e277ea16